src/bidiMapper/utils/log.ts

```diff
@@ -16,17 +16,27 @@
 export function log(type: string): (...message: any[]) => void {
   return (...messages: any[]) => {
     // If run in browser, add debug message to the page.
-    if (typeof window !== 'undefined' && window?.document?.documentElement) {
+    if (globalThis?.document?.documentElement) {
       console.log(type, ...messages);
 
-      const elementId = type + '_log';
+      const typeLogContainer = findOrCreateTypeLogContainer(type);
 
-      if (!window.document.getElementById(elementId)) {
-        window.document.documentElement.innerHTML += `<h3>${type}:</h3><pre id='${elementId}'></pre>`;
-      }
-
-      const element = window.document.getElementById(elementId);
-      element.innerText += messages.join(', ') + '\n';
+      const pre = document.createElement('pre');
+      pre.innerText = messages.join(', ');
```

Comment:
```suggestion
      pre.textContent = messages.join(', ');
```

I would avoid `innerText`, it has very peculiar behavior. `textContent` simply created/updates a single `Text` node, which is much more likely what you want.